### ISSUE TITLE
Publish alpha/beta/rc releases

### ISSUE THREAD

**Issue by ilanbiala** [gulpjs/gulp#865]
I noticed that the 4.0 branch now has 4.0.0-alpha.1 version in the package.json, but that it hasn't been published. Are there any plans to publish alpha/beta/rc releases along the way to 4.0.0?

**Comment by geekflyer** [comment]
:+1:

**Comment by phated** [comment]
This isn't likely.  The reason the 4.0 branch's version was changed was only to work with the new CLI.  npm makes it easy to install from a branch, so there is no real need to publish prereleases to the registry.

**Comment by ilanbiala** [comment]
@phated many more people are more confident in published npm releases than installing from a Git branch. Also, branches can change, npm releases really don't..does it hurt to publish it if you are going to bump the version?

**Comment by geekflyer** [comment]
Agree with @ilanbiala  . Though you can also fix the version by appending a commit hash to the url (e.g. https://github.com/gruntjs/grunt/archive/77155e3f61d213c529a16ea595fc486fdeca28c9.tar.gz) .

Anyway people are usually mentally more comfortable with using an npm release.

Another serious problem I had in the past with including git dependencies, is that npm downloads them using git and then git's own proxy settings apply, instead of the ones from npm.
With npm dependencies I could simply set an "always" proxy, because npm deps are always outside of my corporate network, but with git I have to set mixed rules and excludes (because of our internal repos) and in addition the git protocol is blocked by our corporate proxy. I spent some time to setup it up correctly and now my git simply rewrites git:// urls to https://, but that requires at least git 1.8.4. 
`git config --global url."https://github.com/".insteadOf git@github.com:`

To cut a long story short - npm releases simply work, git dependencies are (sometimes) complicated.

**Comment by sindresorhus** [comment]
> though that doesn't help when someone rewrites the history - deletes the commit. .

Not really true in practise. GitHub preserves it for a long time, as long as you have the SHA.

-

But I do agree npm releases are better for this.

**Comment by knownasilya** [comment]
Yes, please publish an rc, this way it'll be more likely to be accepted and used.

**Comment by ilanbiala** [comment]
@contra @phated @sindresorhus a prime example, this SHA 3eef36ab126e5b858884c44dfbee3a0aecd73fdb used to exist, and now it doesn't seem to resolve properly on GitHub...so please someone publish Gulp#4.0.0-alpha1 and more alphas and betas and rcs....

**Comment by sondr3** [comment]
I've started slowly migrating to Gulp 4.0 and agree, it would be really nice to have some kind of RC/beta releases :+1:

**Comment by nmccready** [comment]
:+1:  +1 , but sha's should be good enough for npm-shrinkwrap correct?

**Comment by danjenkins** [comment]
this should be published to npm as an alpha :+1:

**Comment by pho3nixf1re** [comment]
:+1:

**Comment by contra** [comment]
@phated Any notes as to why we can't publish this to npm? I seem to remember you having a reason on IRC

**Comment by rhutchison** [comment]
@phated @contra :+1:

**Comment by SyntaxRules** [comment]
:+1:

**Comment by phated** [comment]
The 4.0 publish path is going to mirror npm3.  Locking this

**Comment by phated** [comment]
Closing in favor of #1486
